### PR TITLE
Added Rolify

### PR DESCRIPTION
**Pull request by pixeldesu**

### CODE REVIEW

File: spec/models/role_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  end
+
+  it 'should return true if the user has a specified role' do
+    @user.add_role :admin
+    expect(@user.has_role? :admin).to be true
+  end
+
+  it 'should return false if the user does not have a specified role' do
+    @user.add_role :user
+    expect(@user.has_role? :admin).to be false
```
Comment (sup) [inline spec/models/role_spec.rb]: ^

File: config/initializers/rolify.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Rolify.configure do |config|
+  # By default ORM adapter is ActiveRecord. uncomment to use mongoid
+  # config.use_mongoid
+
+  # Dynamic shortcuts for User class (user.is_admin? like methods). Default is: false
+  # config.use_dynamic_shortcuts
+end
```
Comment (sup) [inline config/initializers/rolify.rb]: ^^^^^ important, unix newlines pls

**Comment by sup** [comment]
coverage is :star:, please fix hash syntax and above nits

**Comment by pixeldesu** [comment]
I rather don't touch the migrations, everything Hound is complaining about here is generated.

Also, as mentioned, I don't want to break the default validation syntax, but I'll fix the other cases of old hash syntax.

File: app/models/role.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Role < ActiveRecord::Base
+  has_and_belongs_to_many :users, join_table: :users_roles
+  belongs_to :resource, polymorphic: true
+
+  validates :resource_type,
+            :inclusion => {in: Rolify.resource_types },
```
Comment (pixeldesu) [inline app/models/role.rb]: whoops, I'll fix that one.

**Comment by sup** [comment]
@pixeldesu yeah, i'm gonna add `db/` to hounds ignore list

File: config/initializers/rolify.rb (hunk 2 above)
Comment (sup) [inline config/initializers/rolify.rb]: @pixeldesu this pls

**Comment by pixeldesu** [comment]
@sup, r?

**Comment by sup** [comment]
LGTM :)

**Pull request merged**